homeassistant/components/zwave_js/discovery_data_template.py

```diff
@@ -507,64 +514,57 @@ def values_to_watch(self, resolved_data: dict[str, Any]) -> Iterable[ZwaveValue]
             resolved_data["configuration_value"],
         ]
 
-    def get_speed_config(
+    def get_fan_value_mapping(
         self, resolved_data: dict[str, ZwaveConfigurationValue]
-    ) -> list[int] | None:
-        """Get current speed configuration from resolved data."""
+    ) -> FanValueMapping | None:
+        """Get current fan properties from resolved data."""
         zwave_value: ZwaveValue = resolved_data["configuration_value"]
 
         if zwave_value.value is None:
-            _LOGGER.warning("Unable to read fan speed configuration value")
+            _LOGGER.warning("Unable to read fan configuration value")
             return None
```

Comment:
We shouldn't do this. As far as I can tell, your code doesn't expect this to happen. Instead you should raise `UnknownValueData` as it is handled in the discovery logic: https://github.com/home-assistant/core/blob/95f20500ca8a758c2828b553937fda064740f8a8/homeassistant/components/zwave_js/discovery.py#L916

and the value will be skipped from discovery

Reply:
>  As far as I can tell, your code doesn't expect this to happen.

It does!  The entity's `available` property becomes `false` when this happens.  Check out the earlier discussion thread from the original PR at https://github.com/home-assistant/core/pull/59697#discussion_r753773026.

Reply:
thanks, this is what happens when you come late to the conversation 🤦🏾 . Is creating an entity as unavailable better than just not creating an entity and logging an error? @MartinHjelmare my suggestion differs from your suggested approach, what do you think?

Reply:
If this happens it's most likely a temporary problem that will be solved by interviewing the device again.

Generally we need to cope with the Value.value property returning None.